### PR TITLE
Minor features (NSNetService and nullability fixes)

### PR DESCRIPTION
**Pull request by seanm**
These changes are not as trivial as my last PR, but still pretty straightforward.

### CODE REVIEW

**Comment by seanm** [comment]
@chrisballinger @jakebromberg @jdeff @zhouzhongguang as you've all committed to this project recently, perhaps you'd care to review this too?

File: Source/GCD/GCDAsyncSocket.h [hunk 1]
```diff
@@ -656,7 +662,7 @@ typedef NS_ENUM(NSInteger, GCDAsyncSocketError) {
  * completes writing the bytes (which is NOT immediately after this method returns, but rather at a later time
  * when the delegate method notifies you), then you should first copy the bytes, and pass the copy to this method.
 **/
-- (void)writeData:(NSData *)data withTimeout:(NSTimeInterval)timeout tag:(long)tag;
+- (void)writeData:(nullable NSData *)data withTimeout:(NSTimeInterval)timeout tag:(long)tag;
```
Comment (chrisballinger) [inline Source/GCD/GCDAsyncSocket.h]: When would it be valid to pass nil data here?

Comment (seanm) [reply Source/GCD/GCDAsyncSocket.h]: I made it nullable to conform to the docs, line 651: "If you pass in nil or zero-length data, this method does nothing and the delegate will not be called." There could be code out there relying on this.

Comment (seanm) [reply Source/GCD/GCDAsyncSocket.h]: I'm ok with reverting it to nonnull and changing the docs if you'd prefer...

**Pull request merged**